src/mango/src/mango_idx.erl

```diff
@@ -119,6 +147,16 @@ validate_new(Idx, Db) ->
     Mod:validate_new(Idx, Db).
 
 
+validate_design_opts(Props) ->
+    case lists:keyfind(<<"options">>, 1, Props) of
+        {<<"options">>, {[{<<"partitioned">>, P}]}}
```

Comment:
We are making a dangerous assumption here that options would contain a single element and it will be `{<<"partitioned">>, P}`.